rcsbapi/data/schema.py

```diff
@@ -443,6 +454,31 @@ def get_descendant_fields(self, schema_graph, node, visited=None):
             return result[0]
         return result
 
+    def extract_name_description(self, schema_part, parent_name=""):
+        if isinstance(schema_part, dict):
+            if 'name' in schema_part and 'description' in schema_part:
+                name = schema_part['name']
+                description = schema_part['description']
+                if name in self.seen_names:
```

Comment:
I think with this logic, the first occurrence of a redundant field will be added without a parent_name (like below, 'ChemComp.id' is just under 'id':

![image](https://github.com/user-attachments/assets/6caba823-4006-44aa-ba4d-10b26003831d)

So in this case, if a user was looking for 'ChemComp.id', they would see a field name the doesn't match with the dot notation version. I'm not sure this will exactly work for you, but the way I handled this for `node_index_dict` is I have an attribute `.field_names_list` and I checked the count of a field name to determine whether it was redundant and needed to be in dot notation. None of the types are redundant, so I didn't need to check those.

Reply:
i thought we needed a function that takes a partial string and returns all fields containing that partial string, the dict already has the dot notation for redundant fields.

Reply:
Maybe we can just make a note of it in the PR to master? I think it's ok, it's just that the key just says 'id' even though users need to refer to it as 'ChemComp.id'. I agree that the main purpose of the function is to search, but I also think it's a little confusing if the keys don't match up.